src/animation/AnimationMixer.js

```diff
@@ -81,48 +149,101 @@ THREE.AnimationMixer.prototype = {
 
 	removeAction: function( action ) {
 
-		var index = this.actions.indexOf( action );
+		var actions = this._actions,
+			index = actions.indexOf( action );
 
-		if ( index !== - 1 ) {
+		if ( index === - 1 ) {
 
-			this.actions.splice( index, 1 );
-			action.mixer = null;
+			return this; // we don't know this action - do nothing
 
 		}
 
+		// unreference all property mixers
+		var uuid = ( action.localRoot || this.root ).uuid,
+			actionBindings = action._propertyBindings,
+			bindings = this._bindingsMaps[ uuid ],
 
-		// remove unused property bindings because if we leave them around the mixer can get slow
-		var root = action.localRoot || this.root;
-		var tracks = action.clip.tracks;
+			bindingsChanged = false;
 
-		for ( var i = 0; i < tracks.length; i ++ ) {
+		for( var i = 0, n = actionBindings.length; i !== n; ++ i ) {
 
-			var track = tracks[ i ];
+			var propertyMixer = actionBindings[ i ];
+			actionBindings[ i ] = null;
 
-			var propertyBindingKey = root.uuid + '-' + track.name;
-			var propertyBinding = this.propertyBindingMap[ propertyBindingKey ];
+			// eventually remove the binding from the array
+			if( -- propertyMixer.referenceCount === 0 ) {
 
-			propertyBinding.referenceCount -= 1;
+				propertyMixer.restoreOriginalState();
+				bindingsChanged = true;
 
-			if ( propertyBinding.referenceCount <= 0 ) {
+			}
 
-				propertyBinding.unbind();
+		}
 
-				delete this.propertyBindingMap[ propertyBindingKey ];
+		if ( bindingsChanged ) {
+
+			this._bindingsDirty = true; // invalidates this._bindings
+
+		}
+
+		// remove from array-based unordered set
+		actions[ index ] = actions[ actions.length - 1 ];
+		actions.pop();
```

Comment:
Was inspired by the article @GGAlanSmithee cited above, "suggesting" stuff like `[] + {}` for that :-).